widgetsnbextension/src/manager.js

```diff
@@ -190,12 +188,18 @@ WidgetManager.set_state_callbacks = function (load_callback, save_callback, opti
 var url = [window.location.protocol, '//', window.location.host, window.location.pathname].join('');
 var key = 'widgets:' + url;
 WidgetManager.set_state_callbacks(function() {
-    if (localStorage[key]) {
-        return JSON.parse(localStorage[key]);
-    }
-    return {};
+    return Promise.resolve(Jupyter.notebook.metadata.ipywidgets.state);
 }, function(state) {
-    localStorage[key] = JSON.stringify(state);
+    Jupyter.notebook.metadata.ipywidgets = {
+
+        // Persisted widget state version
+        version: '1.0.0',
```

Comment:
Should this rather be the version of jupyter-js-widgets (or rather of the notebook widget manager)?

I am not sure about this because I think that eventually the extension version info will be in the `_model_module`  and `_view_module` variables.